addons/partner_autocomplete/models/res_company.py

```diff
@@ -1,14 +1,128 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import api, fields, models
+from ast import literal_eval
+from urllib.parse import urlparse
+
+import json
+import base64
+import logging
+import requests
+
+from odoo.addons.iap.tools import iap_tools
+from odoo import fields, models, tools, _
+
+_logger = logging.getLogger(__name__)
+
+COMPANY_AC_TIMEOUT = 5
+
 
 class ResCompany(models.Model):
     _name = 'res.company'
     _inherit = 'res.company'
 
     partner_gid = fields.Integer('Company database ID', related="partner_id.partner_gid", inverse="_inverse_partner_gid", store=True)
+    iap_enrich_auto_done = fields.Boolean('Enrich Done')
 
     def _inverse_partner_gid(self):
         for company in self:
             company.partner_id.partner_gid = company.partner_gid
+
+    def iap_enrich_auto(self):
+        """ Enrich company. This method should be called by automatic processes
+        and a protection is added to avoid doing enrich in a loop. """
+        if self.env.user.has_group('base.group_system'):
```

Comment:
Should be `self.env.is_system()`, unless we specifically don't want this to succeed in sudo(). Also in that case there's `user._is_system()` I guess.